### PR TITLE
Updated documentation on Asset Processor configuration to include some details on each settings registry value.

### PR DESCRIPTION
**Pull request by AMZN-stankowi**
Updated documentation on Asset Processor configuration to include some details on each settings registry value.

Signed-off-by: AMZN-stankowi <4838196+AMZN-stankowi@users.noreply.github.com>

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (amzn-mike) [inline content/docs/user-guide/assets/asset-processor/configuration.md:25]: Might want to add: lower numbers are higher priority, higher numbers are lower priority.

This mostly affects the override system, files in higher priority scanfolders override those in lower priority scanfolders

Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md]: Done

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (sptramer) [inline content/docs/user-guide/assets/asset-processor/configuration.md:11]:
```suggestion
To process assets for a non-host platform, you can enable additional platforms under the settings registry path `Amazon/AssetProcessor/Settings/Platforms`. Note that this is only one step in enabling a platform for your project. See the [platforms](/docs/user-guide/platforms/) section of the documentation for how to set up and deploy to additional platforms.
```

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (sptramer) [inline content/docs/user-guide/assets/asset-processor/configuration.md:15]:
```suggestion
The asset cache server's address can be set using the setting path `Amazon/AssetProcessor/Settings/Server/CacheServerAddress`.
```

Rather than directing users to the Discord, file a GitHub issue requesting documentation for this feature. If you want a reference to the requirement improvement in the document itself, use the `{{< todo issue="###" >}}` shortcode (see [smoketest](https://github.com/o3de/o3de.org/blob/main/content/smoketest.md) for an example). 

Also even if this isn't completely documented, include the required formatting (I'm assuming `<ipv4-address>:<port>` ?)

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (sptramer) [inline content/docs/user-guide/assets/asset-processor/configuration.md:17]:
```suggestion
Metadata files for assets are managed under the setting path `Amazon/AssetProcessor/Settings/MetaDataTypes`, with a format mapping the metadata extension to associated file extension. For example, FBX files use a metadata file type with the extension `.fbx.assetinfo`, which is defined under this setting path as `"fbx.assetinfo": "fbx"`. When adding a new asset type and want to track settings in a metadata file, make sure to add it to this setting path. This is so the Asset Processor will re-run the processing job for the source asset when the metadata file changes.
```

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (sptramer) [inline content/docs/user-guide/assets/asset-processor/configuration.md:27]:
````suggestion
```json
"ScanFolder <scan folder name>": {
  "watch": <scan folder path>",
  "display": "<Display name for the scan folder>",
  "recursive": 0, // 1 scans all subdirectories for assets recursively
  "order": any number // The order this scan folder is handled compared to other scan folders.
}
```
````

Fix formatting to use our standard placeholders, and include information on whether higher-numbered or lower-numbered `order` folders are scanned first.

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (sptramer) [inline content/docs/user-guide/assets/asset-processor/configuration.md:36]:
````suggestion
The exclusion system is used to block files matching patterns from asset processing. These exist under the registry path `Amazon/AssetProcessor/Settings/` with a pattern that looks like:

```json
"Exclude <Exclusion title>": {
  "pattern": "<regex of files to exclude>"
}
```
````

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (sptramer) [inline content/docs/user-guide/assets/asset-processor/configuration.md:43]:
````suggestion
Each platform has a collection of tags applied to it, under the registry path `Amazon/AssetProcessor/Settings/`. These tags are checked by other systems for processing, but don't directly trigger any activity themselves. Platform tags follow the pattern:

```json
"Platform <Platform name>": {
  "tags:" "<comma-separated list of tags>"
}
```
````

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (sptramer) [inline content/docs/user-guide/assets/asset-processor/configuration.md:55]:
````suggestion
Copy jobs are for source assets that should be copied directly to the cache as a product asset. Some assets don't need additional processing, so setting up a copy job this way is quicker than authoring an entire C++ based builder to make content of that type available to your game as an asset. Copy jobs are managed by a registry setting pattern under `Amazon/AssetProcessor/Settings/`, using the following pattern:

```json
"RC <Copy job name>": {
  // Matching type can be glob or pattern, do not use both. Pattern is for regular expression based matching, glob is useful for if you just want to match on a file extension.
  "glob": "*.<extension>",
  "pattern": "<regex pattern to match this copy job>", // It's either glob or pattern, not both
  "productAssetType": "<The UUID for the AZ::Data::AssetType for the product asset>",
  "params": "<copy|skip>", // Use "copy" for copy jobs, "skip" to skip processing. Skip is mostly used for platforms that don't need a particular asset type.
  "platform tag": "<copy|skip>", // Copy or Skip for platforms matching this tag. See the earlier platform tag section for details.
}
```
````

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (sptramer) [inline content/docs/user-guide/assets/asset-processor/configuration.md:59]:
```suggestion
{{< todo issue="706" >}}
To add project-specific or user-specific overrides, you can use the same pattern you would follow with other settings registry overrides. 
{{</todo>}}
```

The `todo` shortcode is provided for linking to open issues which affect the current page.

**Review by sptramer — CHANGES_REQUESTED** [review]
Some minor style changes and a question about how priorities are ranked (low-to-high or high-to-low).

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md]: It's not just documentation we need for it, if someone wants to use this they pretty much have to talk to us directly. Cleaning up this feature is on my team's backlog, but right now you can't really use the feature without having someone on my team walk you through it. It's a 1.x feature that was built for some specific use cases, we want to expand it to support more teams but we haven't gotten around to it.

Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md]: I've added a note on the format. It's not an ip address, it's a network share address.

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md]: I took that change

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md]: I've cleaned up the formatting based on your suggestion

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md]: I've cleaned this up to match your suggestion

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md]: I've updated with this change

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md]: I've updated with this change plus some additional cleanup

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md]: Done

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/configuration.md:9]: Style/editoral suggestion

```suggestion
Many of the content processing rules that **Asset Processor** uses for your project are controlled by settings registry values. You can view the default values in [`Registry/AssetProcessorPlatformConfig.setreg`](https://github.com/o3de/o3de/blob/development/Registry/AssetProcessorPlatformConfig.setreg). This file is well commented, and full of examples you can use for specific configuration on your project's asset pipeline.
```

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (micronAMZN) [inline content/docs/user-guide/assets/asset-processor/configuration.md:15]: Saw the previous discussion on this, one small suggestion.

```suggestion
The Asset Cache server's address can be set using the setting path `Amazon/AssetProcessor/Settings/Server/CacheServerAddress`. In most cases this will be a network share, an example for Windows looks like `T:\\AssetServerCache`. This feature is in development. If you have questions about the Asset Cache server, reach out us on the [O3DE Discord](https://discord.com/invite/o3de).
```

**Review by micronAMZN — APPROVED** [review]
Approving with a couple minor suggestions.

**Comment by micronAMZN** [comment]
Need to fix the sign-off on this PR.

**Comment by AMZN-stankowi** [comment]
> Need to fix the sign-off on this PR.

Fixed, it's here now https://github.com/o3de/o3de.org/pull/1524

File: content/docs/user-guide/assets/asset-processor/configuration.md
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/asset-processor/configuration.md:9]: I've applied these suggested changes, see https://github.com/o3de/o3de.org/pull/1524

**Pull request closed without merging**
